### ISSUE TITLE
.find().where() using search criteria as a field name

### ISSUE THREAD

**Issue by zivc** [balderdashy/sails-mysql#195]
Hello,
My `.find().where()` isn't working properly;
````
sails> Tenant.find().where({'organisation':'test'}).then(console.log);
{ _bitField: 0,
  _fulfillmentHandler0: undefined,
  _rejectionHandler0: undefined,
  _progressHandler0: undefined,
  _promise0: undefined,
  _receiver0: undefined,
  _settledValue: undefined }
sails> TenantID undefined
TenantCode undefined
DataPreference undefined
TenantName undefined
TenantOrganizationDN undefined
AccessScheme undefined
LoadDatePreference undefined
CreatedBy undefined
CreatedDate undefined
ReportDirectory undefined
Unhandled rejection [Error (E_UNKNOWN) Encountered an unexpected error] Details:  Error: ER_BAD_FIELD_ERROR: Unknown column 'test' in 'where clause'
````

Here is my model, https://gist.github.com/zivc/56719fa2b84775afd99e

Using these packages:
sails-mysql v0.10.11
sails v0.11.0

Cheers,

Ash

**Comment by zivc** [comment]
Odd, here is the query that has been generated which looks perfectly fine, but errors when ran in mysql
`SELECT `tenant`.`TenantID`, `tenant`.`TenantCode`, `tenant`.`DataPreference`, `tenant`.`TenantName`, `tenant`.`TenantOrganizationDN`, `tenant`.`AccessScheme`, `tenant`.`LoadDatePreference`, `tenant`.`CreatedBy`, `tenant`.`CreatedDate`, `tenant`.`ReportDirectory` FROM `Tenant` AS `tenant`  WHERE `tenant`.`TenantOrganizationDN` = "test"`

**Comment by zivc** [comment]
The `"test"` should be `'test'`

**Comment by zivc** [comment]
Changing this line temporarily remedied my problems
https://github.com/balderdashy/waterline-sequel/blob/master/sequel/lib/criteriaProcessor.js#L438
from: `value = '"' + utils.escapeString(value) +'"';`
to: `value = '\'' + utils.escapeString(value) +'\'';`

**Comment by zivc** [comment]
More info here, http://dev.mysql.com/doc/refman/5.0/en/sql-mode.html#sqlmode_ansi_quotes

**Comment by devinivy** [comment]
Thanks!  Should a test be failing in sails-mysql/waterline-adapter-tests because of this?  I use sails-mysql, and I've not run into such a (general-seeming) problem.  Is it because of the particular mode you're running mysql in?

**Comment by zivc** [comment]
You need both forks, my sails-mysql and waterline-sequel in sails-mysql module. If the tests are failing i'll take a look.

It's a specific mysql environment I am working in. We're using sails to access a read only slave of a larger, legacy cluster. I think the reason why ANSI_QUOTES is enabled is for the benefit of the queries being interoperable between MySQL and MSSQL.

**Comment by zivc** [comment]
@devinivy I just cloned my fork of `sails-mysql` and replaced the `node_modules/waterline-sequel` with my fork of that and all tests pass.

In `sails-mysql` however, the tests finish with two tests pending?

````Ash$ npm test

> sails-mysql@0.10.11 test /Users/Ash/Projects/smysqlt
> make test

info: Testing `sails-mysql`, a Sails/Waterline adapter.
info: Running `waterline-adapter-tests` against 4 interfaces...
info: ( semantic, queryable, migratable, associations )

debug: Latest draft of Waterline adapter interface spec:
debug: http://links.sailsjs.org/docs/plugins/adapters/interfaces


  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․

  231 passing (2s)
  2 pending

time elapsed: 2602ms````

**Comment by sailsbot** [comment]
Thanks for posting, @zivc.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
